lib/active_resource/base.rb

```diff
@@ -1620,19 +1620,24 @@ def find_or_create_resource_for(name)
         resource_name = name.to_s.camelize
 
         const_args = [resource_name, false]
-        if self.class.const_defined?(*const_args)
-          self.class.const_get(*const_args)
-        else
-          ancestors = self.class.name.to_s.split("::")
-          if ancestors.size > 1
-            find_or_create_resource_in_modules(resource_name, ancestors)
+
+        begin
+          if self.class.const_defined?(*const_args)
+            self.class.const_get(*const_args)
           else
-            if Object.const_defined?(*const_args)
-              Object.const_get(*const_args)
+            ancestors = self.class.name.to_s.split("::")
+            if ancestors.size > 1
+              find_or_create_resource_in_modules(resource_name, ancestors)
             else
-              create_resource_for(resource_name)
+              if Object.const_defined?(*const_args)
+                Object.const_get(*const_args)
+              else
+                create_resource_for(resource_name)
+              end
             end
           end
+        rescue
```

Comment:
Need to tightly scope this rescue so unrelated exceptions aren't captured.

Reply:
Scoped this to just rescue from NameError